plugins/compile.sh

```diff
@@ -7,18 +7,268 @@
+	
+	latest=$(wget -qO- https://www.amxmodx.org/amxxdrop/$version/amxmodx-latest-base-$os)
+	
+	if [ $? -ne 0 ]
+	then
+		echo "Internet connection is required"
+		quit_script
+	fi
+	echo -ne "####                 (20%)\r"
+	wget -q https://www.amxmodx.org/amxxdrop/$version/$latest
+	echo -ne "########             (40%)\r"
+	tar -xzf $latest -C update_files
+	echo -ne "##########           (50%)\r"
+	rm -f $latest
+}
+
+prepare_update () {
+	echo -ne "##################   (90%)\r"
+	dir="update_files/addons/amxmodx"
+	if [ $ver -eq 1 ]
+	then
+		mv $dir/configs/hamdata.ini ../configs/hamdata.ini
+	fi
+	rm -rf $dir/configs
+	rm -rf $dir/logs
+	
+	cp -r $dir ../../
+	echo -ne "#################### (100%)\r"
+}
+
+get_game_files () {
+	if [ -n "$gamemod" ]
+	then 
+		echo -ne "############         (60%)\r"
+		latest=$(wget -qO- https://www.amxmodx.org/amxxdrop/$version/amxmodx-latest-$gamemod-$os)
+		echo -ne "##############       (70%)\r"
+		wget -q https://www.amxmodx.org/amxxdrop/$version/$latest
+		echo -ne "################     (80%)\r"
+		tar -xzf $latest -C update_files
+		rm -f $latest
+	fi
+	
+	prepare_update
+}
+
+update_game () {
+	echo -en "\nUpdating AMXX $version "
+	case $1 in
+		1 )
+			echo -e "Counter-Strike\n"
+			gamemod="cstrike"
+			;;
+		2 )
+			echo -e "Day of Defeat\n"
+			gamemod="dod"
+			;;
+		3 )
+			echo -e "Earth's Special Forces\n"
+			gamemod="esf"
+			;;
+		4 )
+			echo -e "Half-Life\n"
+			;;
+		5 )
+			echo -e "Natural Selection\n"
+			gamemod="ns"
+			;;
+		6 )
+			echo -e "Team Fortress Classic\n"
+			gamemod="tfc"
+			;;
+		7 )
+			echo -e "The Specialists\n"
+			gamemod="ts"
+			;;
+		*)
+	esac
+	
+	echo -ne "##                   (10%)\r"
+	
+	get_base
+	get_game_files $id
+}
+
+get_game () {
+	echo "Please select the Game"
+	echo "  1) Counter-Strike"
+	echo "  2) Day of Defeat"
+	echo "  3) Earth's Special Forces"
+	echo "  4) Half-Life"
+	echo "  5) Natural Selection"
+	echo "  6) Team Fortress Classic"
+	echo "  7) The Specialists"
+	echo "  0) Exit"
+	
+	latest=$(wget -qO- https://www.amxmodx.org/)
+	if [ $? -ne 0 ]
+	then
+		echo "Internet connection is required"
+		quit_script
+	fi
+	
+	echo -n "Number: "
+	read -r id
+	
+	if (( id == 0 )) || (( id > 7 ))
+	then
+		quit_script
+	fi
+	
+	echo -e "\nAMX MOD X Version"
+	echo "  1) 1.9"
+	echo "  2) 1.10"
+	echo -n "Version: "
+	read -r ver
+	
+	if (( ver == 1 ))
+	then
+		version="1.9"
+	elif (( ver == 2 ))
+	then
+		version="1.10"
+	else
+		quit_script
+	fi
+	
+	echo -e "\nBefore updating files, note that all modifications"
+	echo "  of default files will be reset"
+	echo "Default files are shown below"
+	if (( ver == 1 ))
+	then
+		echo "      configs   / hamdata.ini"
+	fi
+	echo "      data      / all default files"
+	echo "      dlls      / all default files"
+	echo "      modules   / all default files"
+	echo "      plugins   / all default files"
+	echo "      scripting / all default files"
+	echo -en "\nContinue ? [Y/n] : "
```

Comment:
Why require a different register?

Reply:
Could you explain what you mean about register?

Reply:
@CrXane 
Yes. This can be useful for debugging, sometimes.
p.s. Write our nicknames on different lines, @WPMGPRoSToTeMa goat )))

Reply:
@Ecklory I thought about adding 1.8.3 & 1.8.2 but breaks it's going to kill the optimization.

Reply:
@CrXane 
In general, there are several options.
1. hlds_run (restart the server anyway).
2. A few more scripts.
3. Rewrite from bash to some golang.

Reply:
@Ecklory
I've updated add added a couple more features. Check the first post.
I'm using bash because it's preinstalled on Unix systems.

Reply:
> I'm using bash because it's preinstalled on Unix systems.

The executable file on the golang requires absolutely nothing and works almost everywhere.

Reply:
@Ecklory
Are you sure?
I'm running Ubuntu 20.04 Desktop and I don't have Go preinstalled.
golang-go requries 324 MB of additional storage and gccgo-go 191 MB.

Reply:
@CrXane 
I'm talking about the fact that hardly anyone will edit the bash script. Since we want functionality, there will only be crutches on bash.

Reply:
@Ecklory
I've made my best to simplify the script and to make it readable on the first glance. The only downside with bash is that the functions have to be created before calling them, which looks like you're looking the code upside-down.